src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests\Compiler;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass;
+use Symfony\Component\DependencyInjection\Compiler\RegisterEnvVarProcessorsPass;
+use Symfony\Component\DependencyInjection\Compiler\ValidateEnvPlaceholdersPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\Extension;
+
+class ValidateEnvPlaceholdersPassTest extends TestCase
+{
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\LogicException
+     * @expectedExceptionMessage Invalid type for env parameter "env(FOO)". Expected "string", but got "bool".
+     */
+    public function testDefaultEnvIsValidatedByType()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('env(FOO)', true);
+        $container->registerExtension(new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'scalar_node' => '%env(FOO)%',
+        ));
+
+        $this->doProcess($container);
+    }
+
+    public function testEnvsAreValidatedInConfig()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('env(NULLED)', null);
+        $container->registerExtension($ext = new EnvExtension());
+        $container->prependExtensionConfig('env_extension', $expected = array(
+            'scalar_node' => '%env(NULLED)%',
+            'int_node' => '%env(int:FOO)%',
+            'float_node' => '%env(float:BAR)%',
+            'bool_node' => '%env(const:BAZ)%',
```

Comment:
perhaps this should be invalid :thinking: it can provide a bool, but also a string or so.

so we need to be a bit more clear on what a placeholder "means" in Config when it has multiple values, that is; i think the must be valid for the current node type. A bool node only expect boolean dummy values for a placeholder.

If it encounters a different type, the config is invalid.